From fc23ccd4b48355f98f99ce2f4b02f6f3f6c3ae7b Mon Sep 17 00:00:00 2001
From: Kevin Darbyshire-Bryant <kevin@darbyshire-bryant.me.uk>
Date: Mon, 4 Jul 2016 20:23:45 +0100
Subject: [PATCH] miniupnpd: add igd2 postrouting table

add MINIUPNPD-POSTROUTING rule. Solves following error

addmasqueraderule() : chain MINIUPNPD-POSTROUTING not found

Signed-off-by: Kevin Darbyshire-Bryant <kevin@darbyshire-bryant.me.uk>
---
 miniupnpd/Makefile               | 2 +-
 miniupnpd/files/firewall.include | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/miniupnpd/Makefile b/miniupnpd/Makefile
index fbbcfee..97f0b6c 100644
--- a/miniupnpd/Makefile
+++ b/miniupnpd/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=miniupnpd
 PKG_VERSION:=2.0
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE_URL:=http://miniupnp.free.fr/files
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
diff --git a/miniupnpd/files/firewall.include b/miniupnpd/files/firewall.include
index 818af9d..0eb96d7 100644
--- a/miniupnpd/files/firewall.include
+++ b/miniupnpd/files/firewall.include
@@ -5,6 +5,7 @@ IP6TABLES=/usr/sbin/ip6tables
 
 iptables -t filter -N MINIUPNPD 2>/dev/null
 iptables -t nat -N MINIUPNPD 2>/dev/null
+iptables -t nat -N MINIUPNPD-POSTROUTING 2>/dev/null
 
 [ -x $IP6TABLES ] && $IP6TABLES -t filter -N MINIUPNPD 2>/dev/null
 
@@ -20,6 +21,7 @@ add_extzone_rules() {
     # IPv4 - due to NAT, need to add both to nat and filter table
     iptables -t filter -I zone_${ext_zone}_forward -j MINIUPNPD
     iptables -t nat -I zone_${ext_zone}_prerouting -j MINIUPNPD
+    iptables -t nat -I zone_${ext_zone}_postrouting -j MINIUPNPD-POSTROUTING
 
     # IPv6 if available - filter only
     [ -x $IP6TABLES ] && {
